### PR TITLE
[FIX] sale_order_type: changes ignored

### PR DESCRIPTION
**Pull request by chienandalu**
f4e61607fae6f735452b735a4cd7f45c5914f71e added some changes that made
imposible to reach a proper write if the sale type didn't have the
proper conditions.

cc @Tecnativa TT29148

### CODE REVIEW

**Comment by pedrobaeza** [comment]
For reducing the diff, why not changing the sequence comparison and remove the `return True` part?

**Comment by chienandalu** [comment]
> For reducing the diff, why not changing the sequence comparison and remove the `return True` part?

Ok, I wanted to simplify it to ease reading. Changed

File: sale_order_type/models/sale_order.py [hunk 1]
```diff
@@ -61,21 +61,24 @@ def create(self, vals):
...
     def write(self, vals):
+        """A sale type could have a different order sequence, so we could
+        need to change it accordingly"""
         if vals.get("type_id"):
             sale_type = self.env["sale.order.type"].browse(vals["type_id"])
             if sale_type.sequence_id:
                 for record in self:
                     if (
                         record.state in {"draft", "sent"}
                         and record.type_id != sale_type
```
Comment (pedrobaeza) [inline sale_order_type/models/sale_order.py]: You can remove this condition.

Comment (chienandalu) [reply sale_order_type/models/sale_order.py]: :+1:  done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
As the bug is important, let's fast-track it:

/ocabot merge patch

**Comment by pedrobaeza** [comment]
As the feature has been introduced, we should fw-port it to recent versions, or someone will miss it.

**Review by HaraldPanten — APPROVED** [review]

**Comment by sergio-teruel** [comment]
How does this affect in v13?

**Comment by pedrobaeza** [comment]
It's already and totally ported in #1553

**Pull request merged**